server/src/main/java/io/spine/server/outbus/CommandOutputBus.java

```diff
@@ -154,38 +134,36 @@ protected abstract boolean validateMessage(Message message,
     @Override
     protected abstract OutputDispatcherRegistry<C, D> createRegistry();
 
-    /**
-     * Validates and posts the message for handling.
-     *
-     * <p>If the message is invalid, the {@code responseObserver} is notified of an error.
-     *
-     * <p>If the message is valid, it {@linkplain #store(Message) may be stored} in the associated
-     * store before passing it to dispatchers.
-     *
-     * <p>The {@code responseObserver} is then notified of a successful acknowledgement of the
-     * passed message.
-     *
-     * @param message          the message to be handled
-     * @param responseObserver the observer to be notified
-     * @see #validateMessage(Message, StreamObserver)
-     */
     @Override
-    public void post(M message, StreamObserver<Response> responseObserver) {
-        checkNotNull(responseObserver);
-
-        final boolean validationPassed = validateMessage(message, responseObserver);
+    protected Optional<IsSent> preProcess(final E message) {
+        final Optional<Throwable> violation = this.validate(message.getMessage());
+        final Optional<IsSent> result = violation.transform(
+                new Function<Throwable, IsSent>() {
+                    @Override
+                    public IsSent apply(@Nullable Throwable input) {
+                        checkNotNull(input);
+                        final Error error = Exceptions.toError(input);
+                        final Status status = Status.newBuilder()
```

Comment:
If you create a utility class (see my comments on `Buses` above), the whole transformation should probably go there.